designate/central/service.py

```diff
@@ -3101,8 +3110,58 @@ def _import(self, context, zone_import, request_body, pool_id=''):
+                        )  # without recordsets object
+                        LOG.debug(f'Founded zone {zone_object}')
+                    except exceptions.ZoneNotFound:
+                        zone = self.create_zone(context, zone)
+                    else:
+                        criterion = {'zone_id': zone_object.id}
+                        rrsets = self.find_recordsets(context, criterion)
+                        actual_rrsets = dict()
+                        for rrset in rrsets:
+                            LOG.debug(f"Checking recordset {rrset} with type {rrset.type} and name {rrset.name}")
+                            if rrset.type == 'SOA':
+                                LOG.debug(f"Ignoring SOA record")
+                                continue
+                            if rrset.type == 'NS' and rrset.name == zone.name:
+                                LOG.debug(f"Ignoring {rrset} with type {rrset.type} and name {rrset.name} for {zone.name}")
+                                continue
+                            LOG.debug(f"Adding recordset {rrset} with type {rrset.type} and name {rrset.name}")
+                            actual_rrsets[f"{rrset.name}_{rrset.type}"] = rrset
+                        LOG.debug(f"Actual recordset {actual_rrsets}")
+                        new_rrsets = {
+                            f"{rrset.name}_{rrset.type}": rrset for rrset in zone.recordsets
+                        }
+                        LOG.debug(f"New recordset {new_rrsets}")
+                        # if record exist in new and old then update
+                        # if record exist in new and not in old then create
+                        # if record not exist in new and exist in old then delete from old
+                        for rrset in actual_rrsets:
+                            recordset_id = actual_rrsets[rrset].id
+                            if rrset not in new_rrsets:
+                                LOG.debug(f"Deleting recordset {recordset_id} for zone {zone_object.id}")
+                                self.delete_recordset(context, zone_object.id, recordset_id)
+                            if rrset in new_rrsets:
+                                new_rrsets[rrset].id = recordset_id
+                                new_rrsets[rrset].zone_id = zone_object.id
+                                new_rrsets[rrset].tenant_id = zone_object.tenant_id
+                                LOG.debug(f"Updating recordset {rrset} for zone {rrset}")
+                                self.storage.update_recordset(context, new_rrsets[rrset])
+                                new_rrsets.pop(rrset)
+                        for rrset in new_rrsets:
+                            if rrset not in actual_rrsets:
+                                LOG.debug(f"Creating recordset {new_rrsets[rrset]} for zone {zone_object.id}")
+                                self.create_recordset(context, zone_object.id, new_rrsets[rrset])
```

Comment:
overall a lot of logic here, please try to split it nicely using additional functions